### ISSUE TITLE
WIKI Updater cannot connect to Azure DevOps WIKI

### ISSUE THREAD

**Issue by rfennell** [rfennell/AzurePipelines#613]
Hi,
I am getting this problem now on my Self-Hosted build VM conneting to an Azure DevOps GIT repo.

I've checked the credentials being used, I've checked the URL of the repo but I keep getting the error.

The YAML in use is (sensitive stuff removed):
```

    -   task: richardfennellBM.BM-VSTS-WIKIUpdater-Tasks.WikiUpdaterTask.WikiUpdaterTask@1
        displayName: 'Git based WIKI Updater (TEST)'
        condition: and(succeeded(), eq(variables['EnvironmentKey'], 'test'))
        inputs:
            repo: 'dev.azure.com/xxxxxx/_git/Documentation'
            filename: ${{ 'ReleaseNotes/Environments/Test/$(Build.BuildNumber).md' }}
            dataIsFile: true
            sourceFile: '$(BUILD.SOURCESDIRECTORY)\GeneratedReleaseNotes.md'
            message: 'Auto-Generated Release Notes'
            gitname: 'Testing Here'
            gitemail: 'test@test.com'
            useAgentToken: false
            user: ${{parameters.wikiUserName}}
            password: ${{parameters.wikiPassword}}
            localpath: $(BUILD.SOURCESDIRECTORY)\Documentation
```
The error stack I get is:

```
2020-01-21T17:28:15.7967590Z ##[section]Starting: Git based WIKI Updater (TEST)
2020-01-21T17:28:15.8174974Z ==============================================================================
2020-01-21T17:28:15.8175086Z Task         : Git based WIKI Updater from Black Marble
2020-01-21T17:28:15.8175147Z Description  : A tools to update or create a WIKI based in a Git repo
2020-01-21T17:28:15.8175218Z Version      : 1.7.5
2020-01-21T17:28:15.8175261Z Author       : Richard Fennell
2020-01-21T17:28:15.8175337Z Help         : Version: 1.7.5. [More Information](https://github.com/rfennell/vNextBuild/wiki/Wiki-Updater-Task/)
2020-01-21T17:28:15.8175408Z ==============================================================================
2020-01-21T17:28:16.0738815Z ##[debug]agent.workFolder=C:\agent\_work
2020-01-21T17:28:16.0832394Z ##[debug]loading inputs and endpoints
2020-01-21T17:28:16.0834139Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2020-01-21T17:28:16.0848567Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2020-01-21T17:28:16.0852005Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2020-01-21T17:28:16.0854726Z ##[debug]loading INPUT_APPENDTOFILE
2020-01-21T17:28:16.0857240Z ##[debug]loading INPUT_DATAISFILE
2020-01-21T17:28:16.0859709Z ##[debug]loading INPUT_FILENAME
2020-01-21T17:28:16.0861557Z ##[debug]loading INPUT_GITEMAIL
2020-01-21T17:28:16.0863586Z ##[debug]loading INPUT_GITNAME
2020-01-21T17:28:16.0865331Z ##[debug]loading INPUT_LOCALPATH
2020-01-21T17:28:16.0867328Z ##[debug]loading INPUT_MESSAGE
2020-01-21T17:28:16.0869082Z ##[debug]loading INPUT_PASSWORD
2020-01-21T17:28:16.0872353Z ##[debug]loading INPUT_REPLACEFILE
2020-01-21T17:28:16.0903210Z ##[debug]loading INPUT_REPO
2020-01-21T17:28:16.0903839Z ##[debug]loading INPUT_SOURCEFILE
2020-01-21T17:28:16.0904156Z ##[debug]loading INPUT_TAGREPO
2020-01-21T17:28:16.0904403Z ##[debug]loading INPUT_USEAGENTTOKEN
2020-01-21T17:28:16.0904957Z ##[debug]loading INPUT_USER
2020-01-21T17:28:16.0905230Z ##[debug]loading SECRET_SYSTEM_ACCESSTOKEN
2020-01-21T17:28:16.0905494Z ##[debug]loaded 18
2020-01-21T17:28:16.1045331Z ##[debug]repo=dev.azure.com/xxxx/_git/Documentation
2020-01-21T17:28:16.1050518Z ##[debug]filename=ReleaseNotes/Environments/Test/2020.01.21.20.md
2020-01-21T17:28:16.1052083Z ##[debug]localpath=C:\agent\_work\14\s\Documentation
2020-01-21T17:28:16.1060417Z ##[debug]contents=null
2020-01-21T17:28:16.1063140Z ##[debug]message=Auto-Generated Release Notes
2020-01-21T17:28:16.1065140Z ##[debug]gitname=Testing Here
2020-01-21T17:28:16.1066714Z ##[debug]gitemail=test@test.com
2020-01-21T17:28:16.1068235Z ##[debug]user=some name here
2020-01-21T17:28:16.1069799Z ##[debug]password=xxxxx
2020-01-21T17:28:16.1072097Z ##[debug]useAgentToken=false
2020-01-21T17:28:16.1073913Z ##[debug]replaceFile=True
2020-01-21T17:28:16.1076128Z ##[debug]appendToFile=True
2020-01-21T17:28:16.1085786Z ##[debug]dataIsFile=true
2020-01-21T17:28:16.1087040Z ##[debug]sourceFile=C:\agent\_work\14\s\GeneratedReleaseNotes.md
2020-01-21T17:28:16.1088831Z ##[debug]tagRepo=False
2020-01-21T17:28:16.1089177Z ##[debug]tag=null
2020-01-21T17:28:16.1105506Z Variable: Repo [dev.azure.com/xxxx/_git/Documentation]
2020-01-21T17:28:16.1105869Z Variable: Filename [ReleaseNotes/Environments/Test/2020.01.21.20.md]
2020-01-21T17:28:16.1106257Z Variable: Contents [null]
2020-01-21T17:28:16.1106489Z Variable: Commit Message [Auto-Generated Release Notes]
2020-01-21T17:28:16.1106898Z Variable: Git Username [Testing Here]
2020-01-21T17:28:16.1107063Z Variable: Git Email [test@test.com]
2020-01-21T17:28:16.1107223Z Variable: Use Agent Token [false]
2020-01-21T17:28:16.1108987Z Variable: Replace File [true]
2020-01-21T17:28:16.1109213Z Variable: Append to File [true]
2020-01-21T17:28:16.1109566Z Variable: Username [xxxx]
2020-01-21T17:28:16.1109731Z Variable: Password [xxxx]
2020-01-21T17:28:16.1110087Z Variable: Localpath [C:\agent\_work\14\s\Documentation]
2020-01-21T17:28:16.1110262Z Variable: Data Is File [true]
2020-01-21T17:28:16.1110420Z Variable: SoureFile [C:\agent\_work\14\s\GeneratedReleaseNotes.md]
2020-01-21T17:28:16.1110594Z Variable: Tag Repo [false]
2020-01-21T17:28:16.1110743Z Variable: Tag [null]
2020-01-21T17:28:16.1336716Z URL used https://xxx:***@dev.azure.com/xxx/_git/Documentation
2020-01-21T17:28:16.1337895Z Cleaned C:\agent\_work\14\s\Documentation
2020-01-21T17:28:16.1353046Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1357852Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1358463Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1358697Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1358905Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1359092Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1359286Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1359459Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1359651Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1359823Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1360013Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1360184Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1360381Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1360562Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1360754Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1361337Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1361546Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1361738Z (node:7836) Warning: Use Cipheriv for counter mode of aes-256-ctr
2020-01-21T17:28:16.1463559Z ##[error]Error: Error: spawn git ENOENT
2020-01-21T17:28:16.1463866Z ##[debug]Processed: ##vso[task.issue type=error;]Error: Error: spawn git ENOENT
2020-01-21T17:28:16.1464429Z     at exports._errnoException (util.js:1020:11)
2020-01-21T17:28:16.1464643Z     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:32)
2020-01-21T17:28:16.1465024Z     at onErrorNT (internal/child_process.js:376:16)
2020-01-21T17:28:16.1465228Z     at _combinedTickCallback (internal/process/next_tick.js:80:11)
2020-01-21T17:28:16.1465452Z     at process._tickCallback (internal/process/next_tick.js:104:9)
2020-01-21T17:28:16.1465624Z     at Module.runMain (module.js:613:11)
2020-01-21T17:28:16.1465823Z     at run (bootstrap_node.js:394:7)
2020-01-21T17:28:16.1465991Z     at startup (bootstrap_node.js:160:9)
2020-01-21T17:28:16.1466171Z     at bootstrap_node.js:507:3
2020-01-21T17:28:16.1466509Z ##[debug]task result: Failed
2020-01-21T17:28:16.1467116Z ##[error]Error: Error: spawn git ENOENT
2020-01-21T17:28:16.1467493Z ##[debug]Processed: ##vso[task.issue type=error;]Error: Error: spawn git ENOENT
2020-01-21T17:28:16.1467709Z     at exports._errnoException (util.js:1020:11)
2020-01-21T17:28:16.1468060Z     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:32)
2020-01-21T17:28:16.1468253Z     at onErrorNT (internal/child_process.js:376:16)
2020-01-21T17:28:16.1468756Z     at _combinedTickCallback (internal/process/next_tick.js:80:11)
2020-01-21T17:28:16.1469718Z     at process._tickCallback (internal/process/next_tick.js:104:9)
2020-01-21T17:28:16.1469886Z     at Module.runMain (module.js:613:11)
2020-01-21T17:28:16.1470067Z     at run (bootstrap_node.js:394:7)
2020-01-21T17:28:16.1470236Z     at startup (bootstrap_node.js:160:9)
2020-01-21T17:28:16.1470412Z     at bootstrap_node.js:507:3
2020-01-21T17:28:16.1497673Z ##[debug]Processed: ##vso[task.complete result=Failed;]Error: Error: spawn git ENOENT
2020-01-21T17:28:16.1497987Z     at exports._errnoException (util.js:1020:11)
2020-01-21T17:28:16.1498169Z     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:32)
2020-01-21T17:28:16.1498368Z     at onErrorNT (internal/child_process.js:376:16)
2020-01-21T17:28:16.1498533Z     at _combinedTickCallback (internal/process/next_tick.js:80:11)
2020-01-21T17:28:16.1498715Z     at process._tickCallback (internal/process/next_tick.js:104:9)
2020-01-21T17:28:16.1499056Z     at Module.runMain (module.js:613:11)
2020-01-21T17:28:16.1499418Z     at run (bootstrap_node.js:394:7)
2020-01-21T17:28:16.1499600Z     at startup (bootstrap_node.js:160:9)
2020-01-21T17:28:16.1499781Z     at bootstrap_node.js:507:3
2020-01-21T17:28:16.1528948Z ##[section]Finishing: Git based WIKI Updater (TEST)
```
The build server itself is an Azure hosted VM; has many things installed to meet the requirements of the build chain (MSBuild, VS, GIT, Docker etc etc.)

The YAML file is a part of a multi-stage pipeline and the above block is from the release phase.

It all seems to work (I think, not executed a release lately) from a visual pipeline; this is the first of the YAML ones.

Any assistance is most welcome.

Cheers

**Comment by rfennell** [comment]
This was logged inside the closed call #497 by @andyofengland - easier to track if it has a dedicated issue

**Comment by rfennell** [comment]
This the YAML from my integration testing pipeline

```
steps:
- task: richardfennellBM.BM-VSTS-WIKIUpdater-Tasks-DEV.WikiUpdaterTask.WikiUpdaterTask@1
  displayName: 'Git based WIKI Updater - Azure DevOps with Creds'
  inputs:
    repo: 'dev.azure.com/richardfennell/Git%20project/_git/Git-project.wiki'
    filename: TestPageWithCreds.md
    contents: |
     Header
     ---
     $(OutputText)
     ---
     Footer
    message: 'Update from Release - $(Release.ReleaseName) with Creds'
    gitname: builduser
    gitemail: 'build@demo'
    user: richardfennell
    password: '$(AzureDevOpsPAT)'
    localpath: '$(System.DefaultWorkingDirectory)\azurerepowithcreds'

```

As noted in the [tasks documentation](https://github.com/rfennell/AzurePipelines/wiki/WIKI-Updater-Tasks) the URL & credentials for Azure DevOps WIKIs can be found as follows

**URL**

The URL to clone a Azure DevOps WIKIs is not obvious. _IT IS NOT THE URL SHOWN IN THE BROWSER WHEN YOU VIEW THE WIKI_ e.g: 

https://dev.azure.com/richardfennell/Git%20project/_wiki/wikis/Git-project.wiki/1/Home

_SO DON'T USE THIS FORM_

To find the URL to clone the repo
1. Load the WIKI in a browser
1. At the top of the menu pane there is a menu (click the elipsis ...)
1. Select the 'Clone repo' option
1. You will get a URL in the form https://richardfennell@dev.azure.com/richardfennell/Git%20project/_git/Git-project.wiki. This is the 
URL needed in the repo parameter of this task without the https://

So yours looks right

**Credentials**

Then for the credentials he ${user} is you organisation account name and the ${password} is your PAT

I can't see if your entries are correct, can you confirm this is the form in use.

**Comment by andyofengland** [comment]
Hi,

I can indeed confirm that the credentials are correct - they are a username and a PAT. The debug messages show the correct values and I’ve tried them directly to assert that the git clone works.

Cheers

Sent from my iPhone

On 21 Jan 2020, at 19:39, Richard Fennell <notifications@github.com> wrote:

﻿

This the YAML from my integration testing pipeline

steps:
- task: richardfennellBM.BM-VSTS-WIKIUpdater-Tasks-DEV.WikiUpdaterTask.WikiUpdaterTask@1
  displayName: 'Git based WIKI Updater - Azure DevOps with Creds'
  inputs:
    repo: 'dev.azure.com/richardfennell/Git%20project/_git/Git-project.wiki'
    filename: TestPageWithCreds.md
    contents: |
     Header
     ---
     $(OutputText)
     ---
     Footer
    message: 'Update from Release - $(Release.ReleaseName) with Creds'
    gitname: builduser
    gitemail: 'build@demo'
    user: richardfennell
    password: '$(AzureDevOpsPAT)'
    localpath: '$(System.DefaultWorkingDirectory)\azurerepowithcreds'



As noted in the tasks documentation<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frfennell%2FAzurePipelines%2Fwiki%2FWIKI-Updater-Tasks&data=02%7C01%7C%7Cb833981db35e4295656e08d79ea9a996%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637152323835144565&sdata=3BHgeiLIW%2BJBFYZ33KAV5gjqtdyQ3yxn20fgiel5KG4%3D&reserved=0> the URL & credentials for Azure DevOps WIKIs can be found as follows

URL

The URL to clone a Azure DevOps WIKIs is not obvious. IT IS NOT THE URL SHOWN IN THE BROWSER WHEN YOU VIEW THE WIKI e.g:

https://dev.azure.com/richardfennell/Git%20project/_wiki/wikis/Git-project.wiki/1/Home<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdev.azure.com%2Frichardfennell%2FGit%2520project%2F_wiki%2Fwikis%2FGit-project.wiki%2F1%2FHome&data=02%7C01%7C%7Cb833981db35e4295656e08d79ea9a996%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637152323835154576&sdata=TiX26XCtF86ELgJrkkcr%2F1KJZz39ConijJS%2FD16DW9s%3D&reserved=0>

SO DON'T USE THIS FORM

To find the URL to clone the repo

  1.  Load the WIKI in a browser
  2.  At the top of the menu pane there is a menu (click the elipsis ...)
  3.  Select the 'Clone repo' option
  4.  You will get a URL in the form https://richardfennell@dev.azure.com/richardfennell/Git%20project/_git/Git-project.wiki<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Frichardfennell%40dev.azure.com%2Frichardfennell%2FGit%2520project%2F_git%2FGit-project.wiki&data=02%7C01%7C%7Cb833981db35e4295656e08d79ea9a996%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637152323835164581&sdata=cjnVxi3pkyGNWyxomVnMcJqKTjS3Ey4qYjtBSKpYgr4%3D&reserved=0>. This is the
URL needed in the repo parameter of this task without the https://

So yours looks right

Credentials

Then for the credentials he ${user} is you organisation account name and the ${password} is your PAT

I can't see if your entries are correct, can you confirm this is the form in use.

**Comment by andyofengland** [comment]
Was your sample above from a multi stage pipeline or from the visual version?

Sent from my iPhone

On 21 Jan 2020, at 19:39, Richard Fennell <notifications@github.com> wrote:

﻿

This the YAML from my integration testing pipeline

steps:
- task: richardfennellBM.BM-VSTS-WIKIUpdater-Tasks-DEV.WikiUpdaterTask.WikiUpdaterTask@1
  displayName: 'Git based WIKI Updater - Azure DevOps with Creds'
  inputs:
    repo: 'dev.azure.com/richardfennell/Git%20project/_git/Git-project.wiki'
    filename: TestPageWithCreds.md
    contents: |
     Header
     ---
     $(OutputText)
     ---
     Footer
    message: 'Update from Release - $(Release.ReleaseName) with Creds'
    gitname: builduser
    gitemail: 'build@demo'
    user: richardfennell
    password: '$(AzureDevOpsPAT)'
    localpath: '$(System.DefaultWorkingDirectory)\azurerepowithcreds'



As noted in the tasks documentation<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Frfennell%2FAzurePipelines%2Fwiki%2FWIKI-Updater-Tasks&data=02%7C01%7C%7Cb833981db35e4295656e08d79ea9a996%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637152323835144565&sdata=3BHgeiLIW%2BJBFYZ33KAV5gjqtdyQ3yxn20fgiel5KG4%3D&reserved=0> the URL & credentials for Azure DevOps WIKIs can be found as follows

URL

The URL to clone a Azure DevOps WIKIs is not obvious. IT IS NOT THE URL SHOWN IN THE BROWSER WHEN YOU VIEW THE WIKI e.g:

https://dev.azure.com/richardfennell/Git%20project/_wiki/wikis/Git-project.wiki/1/Home<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdev.azure.com%2Frichardfennell%2FGit%2520project%2F_wiki%2Fwikis%2FGit-project.wiki%2F1%2FHome&data=02%7C01%7C%7Cb833981db35e4295656e08d79ea9a996%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637152323835154576&sdata=TiX26XCtF86ELgJrkkcr%2F1KJZz39ConijJS%2FD16DW9s%3D&reserved=0>

SO DON'T USE THIS FORM

To find the URL to clone the repo

  1.  Load the WIKI in a browser
  2.  At the top of the menu pane there is a menu (click the elipsis ...)
  3.  Select the 'Clone repo' option
  4.  You will get a URL in the form https://richardfennell@dev.azure.com/richardfennell/Git%20project/_git/Git-project.wiki<https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Frichardfennell%40dev.azure.com%2Frichardfennell%2FGit%2520project%2F_git%2FGit-project.wiki&data=02%7C01%7C%7Cb833981db35e4295656e08d79ea9a996%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637152323835164581&sdata=cjnVxi3pkyGNWyxomVnMcJqKTjS3Ey4qYjtBSKpYgr4%3D&reserved=0>. This is the
URL needed in the repo parameter of this task without the https://

So yours looks right

Credentials

Then for the credentials he ${user} is you organisation account name and the ${password} is your PAT

I can't see if your entries are correct, can you confirm this is the form in use.

**Comment by rfennell** [comment]
It was a YAML export from a UI based pipeline.

Just to confirm it is not a user name & a PAT for Azure DevOps but the name of the Azure DevOps Organisation and a PAT - confusing or what !

Other option is to tick the box to allow the creds of the build agent to be used to access the WIKI assuming you grant them rights

**Comment by andyofengland** [comment]
It might be worth sticking it in a muti-stage pipeline for completeness...  I've taken the YAML above from a UI version to make sure it was correct.

Organisation & PAT? Wow that's confusing...  Even though it works on the command line with my username and my PAT?

I'd like to use the OAUTH checkbox but I've yet to work out how to set that using the YAML.

**Comment by rfennell** [comment]
My understanding is you need for form

```
steps:
- checkout: self
  clean: true
  persistCredentials: true
```

To get the OAUTH available to tasks see [here](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#checkout)

I have not seen any tasks that work in the UI but not YAML personally, but will give it a try when I get a chance

**Comment by andyofengland** [comment]
Hmm, I don't think the OAUTH bit will work with a multi-stage pipeline as the YAML for release doesn't ever use the explicit source code "get" and may be on a totally different agent.  Back to the credentials I think.

I'll have a try with the organisation and the personal PAT to see if that has any impact

**Comment by andyofengland** [comment]
Re those Cipher errors; it's not something along these lines is it?  Could a NodeJS difference be the problem?

https://stackoverflow.com/questions/55091341/node-cipheriv-warning-in-azure-devops-extension-release-task

**Comment by rfennell** [comment]
OK I have tried this task in a multistage YAML using

1. In a stage straight after a checkout that is set to persist credentials then the task is set to use OUATH
1. In a stage on it's own using OAUTH
1. Using passed creds (instance name and PAT)

The all work, when I pass a non-expired PAT ! (that wasted a good few minutes)

The YAML I used is as follows. It is writing to a WIKI in the same Azure DevOps team project, but did try cross project and that was fine too (assuming the account had rights)


```
trigger:
- master

pool:
  vmImage: 'windows-latest'

stages:
- stage: 'Stage1'
  jobs:
  - job: 'GetSource'
    timeoutInMinutes: 0
    steps:
    - checkout: self
      persistCredentials: true
      clean: true
    - task: richardfennellBM.BM-VSTS-WIKIUpdater-Tasks-DEV.WikiUpdaterTask.WikiUpdaterTask@1
      displayName: 'Git based WIKI Updater - Azure DevOps as Agent after checkout'
      inputs:
        repo: 'dev.azure.com/richardfennell/GitHub/_git/GitHub.wiki'
        filename: TestPageForWikiUpdaterYAMLCheckout.md
        contents: |
          Header
          ---
          $(build.buildnumber)
          ---
          Footer
        message: 'Update from Release - $(build.buildnumber) as Agent'
        gitname: builduser
        gitemail: 'build@demo'
        useAgentToken: true
        localpath: '$(System.DefaultWorkingDirectory)\azurerepoasagentyaml1'

- stage: 'Stage2'
  jobs:
  - job: 'GetSource'
    timeoutInMinutes: 0
    steps:
    - task: richardfennellBM.BM-VSTS-WIKIUpdater-Tasks-DEV.WikiUpdaterTask.WikiUpdaterTask@1
      displayName: 'Git based WIKI Updater - Azure DevOps as Agent no checkout'
      inputs:
        repo: 'dev.azure.com/richardfennell/GitHub/_git/GitHub.wiki'
        filename: TestPageForWikiUpdaterYAMLNoCheckout.md
        contents: |
          Header
          ---
          $(build.buildnumber)
          ---
          Footer
        message: 'Update from Release - $(build.buildnumber) as Agent'
        gitname: builduser
        gitemail: 'build@demo'
        useAgentToken: true
        localpath: '$(System.DefaultWorkingDirectory)\azurerepoasagentyaml2'


- stage: 'Stage3'
  jobs:
  - job: 'WIKIUpdate'
    timeoutInMinutes: 0
    steps:
    - task: richardfennellBM.BM-VSTS-WIKIUpdater-Tasks-DEV.WikiUpdaterTask.WikiUpdaterTask@1
      displayName: 'Git based WIKI Updater - Azure DevOps with Creds'
      inputs:
        repo: 'dev.azure.com/richardfennell/GitHub/_git/GitHub.wiki'
        filename: TestPageYAMLWithCreds.md
        contents: |
          Header
          ---
          $(build.buildnumber)
          ---
          Footer
        message: 'Update from Release - $(build.buildnumber) with Creds'
        gitname: builduser
        gitemail: 'build@demo'
        user: richardfennell
        password: '$(AzureDevOpsPAT)'
        localpath: '$(System.DefaultWorkingDirectory)\azurerepowithcredsyaml'

```

Hope this helps

**Comment by rfennell** [comment]
As per my standard practice I am closing this issue as I have heard nothing for a while. If this issue remains a problem please do not hesitate to re-open it

**Comment by NightFox7** [comment]
@rfennell I am having problems with git clone authentication. After hours of research, I concluded that I need to pass this argument `-c http..extraheader="AUTHORIZATION: bearer ***"` to the git clone command. Will it be possible to add an optional argument that allow this ?

**Comment by rfennell** [comment]
That's not impossible, but a couple of questions

1. What is the *** in the extra header, is it the PAT?
2. Any suggestion how I an repo the issue to make sure any change works before I ship, or to put it another way wgat is strange about your setup?

**Comment by NightFox7** [comment]
Thank you for the quick response.

  1. The *** is the $(system.accesstoken)
  2. Unfortunately, The repo of our organization is not on Azure Cloud neither exposed over internet so I don't know how you can test this

**Comment by NightFox7** [comment]
Looking at you code, I might suggest something like this:
``` js
try {
    await git.silent(true).clone(remote, localpath);
} catch (error) {
    logError(error);
    logDebug(`Trying clone with extraheader`);
    await git.silent(true).clone(`https://${repo}`, localpath, [`-c http.extraheader="AUTHORIZATION: bearer ${password}"`]);
}
```

**Comment by rfennell** [comment]
Yes, I ended up with the same plan. I need to wrapper in flags so the default behaviour is unchanged.

**Comment by rfennell** [comment]
Reopen to track the change

**Comment by rfennell** [comment]
This has been fixed by optionally passing the credentials in the header as opposed to the url.

See #625 for the details.

It has been shipped as 1.8.7
